### PR TITLE
Stress fix

### PR DESCRIPTION
**Pull request by bindgens1**
Fixes #

Description of changes:
 - 

PR Checklist
------------
 - [x] Tests?
   - [x] Interface
   - [x] Core 
 - [ ] Docs?

### CODE REVIEW

**Comment by fweik** [comment]
You have your build directory checked in, please don't do that.

**Comment by mkuron** [comment]
Also, after you remove the build directory, squash your git history (`git rebase -i python`, replace all except for the first "pick" with "squash") so that the Espresso repo doesn't grow by 60MB after your PR is merged...

**Comment by rempferg** [comment]
Chill out. This is not at all merge ready. No idea how you came up with that ;)

**Comment by RudolfWeeber** [comment]
Could you please try lj_force() from test_common instead of the custom eqn in your test?
That variant is tested against the core. See the verify_lj_forces() in tests_common for usage

**Comment by rempferg** [comment]
Wishlist test case:
- separate with and without LE into two test functions with different ut.skipIf
- add a second particle pair or change particle type once to check nonbonded stress inter and -intra
- if it exists, check stress per bond type

File: src/core/ghosts.cpp [hunk 1]
```diff
@@ -36,6 +36,7 @@
 #include "communication.hpp"
 #include "grid.hpp"
 #include "particle_data.hpp"
+#include "domain_decomposition.hpp"
```
Comment (fweik) [inline src/core/ghosts.cpp:39]: Why is this needed?

File: src/core/pressure_inline.hpp [hunk 2]
```diff
@@ -296,14 +296,34 @@ inline void add_bonded_virials(Particle *p1)
       return;
     }
 
-    get_mi_vector(dx, p1->r.p, p2->r.p);
```
Comment (fweik) [inline src/core/pressure_inline.hpp:299]: You can do just
```c++
auto dx = get_mi_vector(p1->r.p, p2->r.p);
```

File: src/core/pressure_inline.hpp [hunk 3]
```diff
@@ -296,14 +296,34 @@ inline void add_bonded_virials(Particle *p1)
...
+    double a[3] = {p1->r.p[0], p1->r.p[1], p1->r.p[2]};
+    double b[3] = {p2->r.p[0], p2->r.p[1], p2->r.p[2]};
+    
+    //printf("Coordinates part 1: %f, %f, %f \n", a[0], a[1], a[2]);
```
Comment (fweik) [inline src/core/pressure_inline.hpp]: Please remove commented out code.

File: src/core/pressure_inline.hpp [hunk 4]
```diff
@@ -296,14 +296,34 @@ inline void add_bonded_virials(Particle *p1)
...
+    //printf("Coordinates part 2: %f, %f, %f \n", b[0], b[1], b[2]);
+
+    double n_le_shifts = dround((a[1]-b[1]) * box_l_i[1]);
```
Comment (fweik) [inline src/core/pressure_inline.hpp]: This should go into the ifdefed block I guess.

File: testsuite/python/lees_edwards_stress.py [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+    print(sim_stress_nonbonded)
+    print('analytic non-bonded stress=')
+    print(anal_stress_nonbonded)
+
+    print('\nsimulated total stress=')
+    print(sim_stress_total)
+    print('analytic total stress=')
+    print(anal_stress_total)
+
+    self.assertTrue(np.max(np.abs(sim_stress_ideal-anal_stress_ideal)) < tol, 'ideal stress does not match analytical result')
```
Comment (fweik) [inline testsuite/python/lees_edwards_stress.py]: Apparently the current style is to use assertLess and so on, because it produces more helpful error messages than "Fals is not True".

**Review by fweik — CHANGES_REQUESTED** [review]
I suggested some smaller clean ups, otherwise this look good to me. Or at least better than before...

File: src/core/pressure_inline.hpp (hunk 4 above)
Comment (rempferg) [reply src/core/pressure_inline.hpp]: done

File: src/core/pressure_inline.hpp (hunk 3 above)
Comment (rempferg) [reply src/core/pressure_inline.hpp]: done

File: src/core/pressure_inline.hpp (hunk 2 above)
Comment (rempferg) [reply src/core/pressure_inline.hpp:299]: done

File: src/core/ghosts.cpp (hunk 1 above)
Comment (rempferg) [reply src/core/ghosts.cpp:39]: Because dd is used in sections ifdef'd with LEES_EDWARDS:
https://github.com/bindgens1/espresso/blob/c20273e50ab7b3707a6e9aab93250c2c2478e5d2/src/core/ghosts.cpp#L340
https://github.com/bindgens1/espresso/blob/c20273e50ab7b3707a6e9aab93250c2c2478e5d2/src/core/ghosts.cpp#L343
https://github.com/bindgens1/espresso/blob/c20273e50ab7b3707a6e9aab93250c2c2478e5d2/src/core/ghosts.cpp#L469
https://github.com/bindgens1/espresso/blob/c20273e50ab7b3707a6e9aab93250c2c2478e5d2/src/core/ghosts.cpp#L472

**Comment by fweik** [comment]
Please make one round of autopep8

**Review by fweik — APPROVED** [review]

**Review by rempferg — APPROVED** [review]

**Comment by bindgens1** [comment]
Hallo Georg,

Danke fürs mergen. Habe gesehen, dass iwo die Änderungen an der Virtuellen
Bindung verloren gegangen sind. Ich werde die nochmal im aktuellen Python
branch zufügen und einen Pullrequest stellen. Da das nur eine kleine
Änderung ist, sollte das unkompliziert gehen. Außerdem muss ich lernen, wie
das geht!

Gruß,
Sebastian

Georg Rempfer <notifications@github.com> schrieb am Di. 26. Sep. 2017 um
10:10:
